impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -1624,6 +1642,7 @@ private static boolean eq(Object o1, Object o2) {
 
     private volatile ServiceProvider<Service> serviceProvider;
     private final Map<Store<?, ?>, List<Copier>> createdStores = new ConcurrentWeakIdentityHashMap<Store<?, ?>, List<Copier>>();
+    private final Map<OnHeapStore<?, ?>, Collection<TierOperationStatistic<?, ?>>> tierOperationStatistics = new ConcurrentWeakIdentityHashMap<OnHeapStore<?, ?>, Collection<TierOperationStatistic<?, ?>>>();
```

Comment:
Whatever is being put in this map is never removed. And from the previous consequence of registering associations with the `StatisticsManager` I wonder if the operations contained in the value do not need to be de-associated.
This comment applies to all `Store.Provider` implementations that now have a `tierOperationStatistics` field.